fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMultiDistinctRule.java

```diff
@@ -54,13 +60,21 @@ public RewriteMultiDistinctRule() {
     public boolean check(OptExpression input, OptimizerContext context) {
         LogicalAggregationOperator agg = (LogicalAggregationOperator) input.getOp();
 
+        // any aggregate function is distinct and constant, we replace it to any_value
+        if (isComplexConstantCountDistinct(input)) {
+            return true;
+        }
+
         Optional<List<ColumnRefOperator>> distinctCols = Utils.extractCommonDistinctCols(agg.getAggregations().values());
 
         // all distinct function use the same distinct columns, we use the split rule to rewrite
-        return !distinctCols.isPresent();
+        return distinctCols.isEmpty();
     }
 
     public List<OptExpression> transform(OptExpression input, OptimizerContext context) {
+        if (isComplexConstantCountDistinct(input)) {
```

Comment:
How to process `count(distinct row(1, 2), count(distinct col1), count(distinct col2)`?

Reply:
will rewrite to `any_value(row(1, 2) IS NULL), count(distinct col1), count(distinct col2)`, other will use default strategy